backends-clickhouse/pom.xml

```diff
@@ -79,7 +79,7 @@
     <dependency>
       <groupId>org.scalacheck</groupId>
       <artifactId>scalacheck_${scala.binary.version}</artifactId>
-      <version>1.13.5</version>
+      <version>1.14.3</version>
```

Comment:
https://github.com/oap-project/gluten/blob/d843bd93b460d6988c5d654d57978d096b0a835a/backends-clickhouse/pom.xml#L106-L117

There are some related dependencies that use hard coding `_ 2.12` in this `pom.xml`, they should also use `${scala.binary.version}`, I suspect there are other similar cases